nvi-rest/src/main/java/no/sikt/nva/nvi/rest/upsert/UpdateNviCandidateStatusHandler.java

```diff
@@ -82,14 +81,10 @@ protected Integer getSuccessStatusCode(NviStatusRequest input, CandidateDto outp
     return HTTP_OK;
   }
 
-  private Candidate fetchCandidate(UUID candidateIdentifier) {
-    return Candidate.fetch(() -> candidateIdentifier, candidateRepository, periodRepository);
-  }
-
   private Candidate updateAndRefetch(
       Candidate candidate, UpdateStatusRequest updateRequest, UserInstance userInstance) {
-    candidate.updateApprovalStatus(updateRequest, userInstance);
-    return fetchCandidate(candidate.getIdentifier());
+    approvalService.updateApproval(candidate, updateRequest, userInstance);
+    return candidateService.getByIdentifier(candidate.identifier());
   }
```

Comment:
DynamoDB is eventually consistent by default, meaning read requests might not return the most recently written data for a short period of time. Other places in application we return a data we have written to the table on success response from DynamoDb, without refetching it. If you want to secure that you actually return the update data, you have to refetch till you get back updated image.

Reply:
All read operations should be using the `consistentRead=true` flag now, so this should not be an issue.